### PR TITLE
Add --to-addr option when claiming GAS

### LINKED ISSUE #755: Support --change-addr and --to-addr for claim command
### Current behavior
The `claim` command does not support specifying to which address the claim is send, nor to which address potential change is send. 

### Expected behavior
We are refactoring the CLI commands according to this issue (#623) in the `refactor-prompt` branch. Once that issue is resolved we should enhance the `claim` command to support the above 2 parameters.

### PR DESCRIPTION
**Pull request by jseagrave21**
**What current issue(s) does this address, or what feature is it adding?**
- Resolves #755 
- Adds a --to-addr option when claiming GAS
  - the --to-addr supercedes the default change address and the from-addr (if specified)

**NOTE:** Please see remaining question regarding issue here https://github.com/CityOfZion/neo-python/issues/755#issuecomment-451917260

**How did you solve this problem?**
Trial and Error

**How did you make sure your solution works?**
Unittest

**Are there any special changes in the code that we should be aware of?**
No

**Please check the following, if applicable:**

- [x] Did you add any tests?
- [x] Did you run `make lint`?
- [x] Did you run `make test`?
- [x] Are you making a PR to a feature branch or development rather than master?
- [x] Did you add an entry to `CHANGELOG.rst`? (if not, please do)

### CODE REVIEW

**Comment by ixje** [comment]
>  I think what they are calling the "change address" is actually what we are calling the --to-addr. What do you think, @ixje?

I swear I thought I already commented on the issue, but apparently not. To answer: I don't know. I can't tell from the code linked and I don't want to dig deeper.

File: neo/Prompt/Commands/Wallet.py [hunk 1]
```diff
@@ -196,16 +196,19 @@ def __init__(self):
...
             args, from_addr_str = PromptUtils.get_from_addr(args)
+            args, to_addr_str = PromptUtils.get_to_addr(args)
 
-        return ClaimGas(PromptData.Wallet, True, from_addr_str)
+        return ClaimGas(PromptData.Wallet, True, from_addr_str, to_addr_str)
 
     def command_desc(self):
         p1 = ParameterDesc('--from-addr', 'source address to claim gas from (if not specified, take first address in wallet)', optional=True)
-        return CommandDesc('claim', 'claim gas', params=[p1])
+        p2 = ParameterDesc('--to-addr', 'destination address for claimed gas (if not specified, use the default change address; or, use the from address, if specified)', optional=True)
```
Comment (ixje) [inline neo/Prompt/Commands/Wallet.py]:
```suggestion
        p2 = ParameterDesc('--to-addr', 'destination address for claimed gas (if not specified, take first address in wallet; or, use the from address, if specified)', optional=True)
```
let's use the same phrasing as `p1`

File: neo/Prompt/Commands/Wallet.py [hunk 2]
```diff
@@ -317,6 +320,9 @@ def ClaimGas(wallet, require_password=True, from_addr_str=None):
         claim_tx.Attributes = [TransactionAttribute(usage=TransactionAttributeUsage.Script,
                                                     data=standard_contract.Data)]
 
+    if to_addr_str:
+        script_hash = wallet.ToScriptHash(to_addr_str)
```
Comment (ixje) [inline neo/Prompt/Commands/Wallet.py]: this lacks error checking
```
neo> wallet claim --to-addr bla
Could not execute command: subsection not found
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/threading.py", line 884, in _bootstrap
    self._bootstrap_inner()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/erik/Documents/code/test-issues/seagrave/venv/lib/python3.6/site-packages/twisted/_threads/_threadworker.py", line 46, in work
    task()
  File "/Users/erik/Documents/code/test-issues/seagrave/venv/lib/python3.6/site-packages/twisted/_threads/_team.py", line 190, in doWork
    task()
  File "/Users/erik/Documents/code/test-issues/seagrave/venv/lib/python3.6/site-packages/twisted/python/threadpool.py", line 250, in inContext
    result = inContext.theWork()
  File "/Users/erik/Documents/code/test-issues/seagrave/venv/lib/python3.6/site-packages/twisted/python/threadpool.py", line 266, in <lambda>
    inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
  File "/Users/erik/Documents/code/test-issues/seagrave/venv/lib/python3.6/site-packages/twisted/python/context.py", line 122, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/Users/erik/Documents/code/test-issues/seagrave/venv/lib/python3.6/site-packages/twisted/python/context.py", line 85, in callWithContext
    return func(*args,**kw)
  File "neo/bin/prompt.py", line 216, in run
    traceback.print_stack()
Traceback (most recent call last):
  File "neo/bin/prompt.py", line 202, in run
    cmd.execute(arguments)
  File "/Users/erik/Documents/code/test-issues/seagrave/neo/Prompt/Commands/Wallet.py", line 68, in execute
    return self.execute_sub_command(item, arguments[1:])
  File "/Users/erik/Documents/code/test-issues/seagrave/neo/Prompt/CommandBase.py", line 75, in execute_sub_command
    return self.__sub_commands[id].execute(arguments)
  File "/Users/erik/Documents/code/test-issues/seagrave/neo/Prompt/Commands/Wallet.py", line 206, in execute
    return ClaimGas(PromptData.Wallet, True, from_addr_str, to_addr_str)
  File "/Users/erik/Documents/code/test-issues/seagrave/neo/Prompt/Commands/Wallet.py", line 324, in ClaimGas
    script_hash = wallet.ToScriptHash(to_addr_str)
  File "/Users/erik/Documents/code/test-issues/seagrave/neo/Wallets/Wallet.py", line 849, in ToScriptHash
    data = b58decode(address)
  File "/Users/erik/Documents/code/test-issues/seagrave/venv/lib/python3.6/site-packages/base58.py", line 96, in b58decode
    acc = b58decode_int(v)
  File "/Users/erik/Documents/code/test-issues/seagrave/venv/lib/python3.6/site-packages/base58.py", line 83, in b58decode_int
    decimal = decimal * 58 + alphabet.index(char)
ValueError: subsection not found
neo>
```

**Comment by jseagrave21** [comment]
@ixje Okay, this should be ready for another review. I used the error checking from `Send.py`.

**Comment by jseagrave21** [comment]
@ixje instead of completely substituting with `neocore.utils.isvalidpublicaddress`, I incorporated the first check into the `ToScripthash` function in `Wallets/Wallet.py`. I also, like that this way provides better feedback than a boolean

**Comment by ixje** [comment]
I don't understand the substitution part 🤔 . My idea was something along the lines of
```python
if args:
    args, from_addr_str = PromptUtils.get_from_addr(args)
    if not isValidPublicAddress(from_addr_str):
        print("Invalid --from-address")
        return False
    
    args, to_addr_str = PromptUtils.get_to_addr(args)
    if not isValidPublicAddress(from_addr_str):
        print("Invalid --to-addr")
        return False

    return ClaimGas(PromptData.Wallet, True, from_addr_str)
```

I'm fine with your approach as well. It's more detailed 👍

**Pull request merged**